lib/web_ui/lib/src/engine/view_embedder/application_dom/application_dom.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../dom.dart';
+
+import '../hot_restart_cache_handler.dart';
+import 'custom_element_application_dom.dart';
+import 'full_page_application_dom.dart';
+
+/// Provides the API that the FlutterViewEmbedder uses to interact with the DOM.
+///
+/// The base class handles "global" stuff that is shared across implementations,
+/// like handling hot-restart cleanup.
+///
+/// This class is specialized to handle different types of DOM embeddings:
+///
+/// * [FullPageApplicationDom] - The default behavior, where flutter takes
+///   control of the whole web page. This is how Flutter Web used to operate.
+/// * [CustomElementApplicationDom] - Flutter is rendered inside a custom host
+///   element, provided by the web app programmer through the engine
+///   initialization.
+abstract class ApplicationDom {
```

Comment:
Bikeshed time! I'm not sure `ApplicationDom` is the best choice because both "application" and "DOM" are global concepts. I think the name should link more directly to the [FlutterView](https://api.flutter.dev/flutter/dart-ui/FlutterView-class.html) class. Ideas:

* `FlutterWebView`
* `WebEngineView`
* `EngineView` (just because we have a lot of `EngineBlah` classes already)

Implementations can be `FullPageView` and `CustomElementView`.

Reply:
What about `GlassPaneHost`? I was thinking that this class would only own the injection of the GlassPane element into the DOM (and whatever config is required for it to work, like the CSS tweaks and all that jazz).

Reply:
I know this ship may have sailed by now, but "glasspane" is kind of a misnomer. I would avoid propagating this name. It used to refer to the element sitting on top of the canvas that captures pointer events (like a glasspane on a picture frame), but now it has a different role entirely even though it still bears the old name. I would still go with "view" somewhere in the name because it's exactly what each element is for.

Reply:
If/when we get the chance, we should rename `<flt-glass-pane>` to `<flutter-view>`.

Reply:
It makes sense, I removed some of the comments about the position of the glasspane in the sandwich of the flutter app, which were no longer true. I just think that the word "View" is almost meaningless (too many uses) in our codebase :P

Reply:
> We should rename `<flt-glass-pane>` to `<flutter-view>`

It's a [trivial change](https://github.com/flutter/engine/blob/7e03ebcbee0dfb4d361618deeeb3d4595156ca65/lib/web_ui/lib/src/engine/embedder.dart#L50-L51), however it has the potential of [breaking the internet](https://github.com/search?q=%22flt-glass-pane%22&type=code)?

Reply:
Will be followed up here: https://github.com/flutter/flutter/issues/117034